pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -470,9 +470,17 @@ func (r *Reconciler) createVirtualMachine(ctx context.Context, nicName string) e
 
 	vmInterface, err := r.virtualMachinesSvc.Get(ctx, vmSpec)
 	if err != nil && vmInterface == nil {
-		vmZone, zoneErr := r.getVirtualMachineZone(ctx)
-		if zoneErr != nil {
-			return errors.Wrap(zoneErr, "failed to get availability zone")
+		var vmZone string
+		var zoneErr error
+
+		vmZone = r.scope.MachineConfig.AZ
+
+		if r.scope.MachineConfig.AZ == "" {
+			klog.Info("No availability zone set, seletecting random availability zone")
```

Comment:
We could move this log line after the error message and include the selected zone